sql/executor.go

```diff
@@ -83,7 +83,15 @@ func newExecutor(db client.DB, gossip *gossip.Gossip, leaseMgr *LeaseManager) *E
 		leaseMgr: leaseMgr,
 	}
 	exec.systemConfigCond = sync.NewCond(&exec.systemConfigMu)
-	gossip.RegisterSystemConfigCallback(exec.updateSystemConfig)
+
+	configC := make(chan *config.SystemConfig, 4)
```

Comment:
A `chan *SystemConfig` can break down if `updateSystemConfig` is so slow that it can't keep up. Adding a buffer postpones the time when this is noticeable, but doesn't prevent the issue from occurring. If the channel gets full, the gossip system gets blocked. Less dramatically, whenever the channel is non-empty it will replay all received `SystemConfigs` in order, which all we really need is to skip to the latest. 

I would suggest replacing the `chan *SystemConfig` with an `chan struct{}` with a buffer size of 1, which simply signals that new information is available (since the caller doesn't need to choose a buffer size, the channel can be returned from `RegisterSystemConfigChannel` instead of being passed in). Always use a non-blocking send, since there is no need to queue up more than one update. After reading from the channel, call `gossip.SystemConfig()` (or whatever) to get the latest information.